src/common/store/background-epics.ts

```diff
@@ -50,11 +58,21 @@ export const annotationLocateEpic: StandardEpic = (action$, state$) => action$.p
   )),
 );
 
+// TODO better return more than one action in annotationLocateEpic (not sure how to do it)
+export const annotationStageEpic: StandardEpic = (action$, state$) => action$.pipe(
+  ofType(LOCATE_ANNOTATIONS),
+  map((action) => {
+      const newAction = setAnnotationStage(AnnotationsStage.located);
+      return syncTabMark(action, newAction);
+    },
+  ),
+);
+
```

Comment:
This task to emit more than one action looked interesting, so I solved it now:
```
export const annotationLocateEpic: StandardEpic = (action$, state$) => action$.pipe(
  ofType('API_READ'),
  filter(action => getActionResourceType(action) === resourceTypes.ANNOTATIONS),
  mergeMap(action =>
    defer(async () => {
        const tabId = retrieveLogicalActionTab(action, state$.value.tabs);
        setBadgeLocating(tabId);
        const locationData = await tabLocateAnnotations(tabId, action.payload.data);
        syncBadgeWithAnnotations(locationData, tabId);
        return [
          locateAnnotations(locationData), 
          setAnnotationStage(AnnotationsStage.located)
        ].map(
          newAction => syncTabMark(action, newAction)
        )
      },
    )
  ),
  // mergeMap treats array as a stream, every element is emitted separately
  mergeMap(action => action)
);
```

Reply:
Thanks, that's cool!